Bottom: a1a14c90cd6483f9dd4fc5f02a8b4fae59b94b8c
Top:    49d1adb95dd754335764533f0d55ed3d5abd4c34
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-10 19:48:43 +0000

Refresh of attempt-to-fix-slowness

---

diff --git a/src/polygon1.c b/src/polygon1.c
index c8a432d..d99984d 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1991,6 +1991,13 @@ poly_Boolean_free (POLYAREA * ai, POLYAREA * bi, POLYAREA ** res, int action)
       M_B_AREA_Collect (&e, b, res, &holes, action);
       poly_Free (&b);
 
+      /* delete of a_isected which is left */
+      while ((p = a_isected) != NULL)
+      {
+        a_isected = p->next;
+        poly_DelContour (&p);
+      }
+
       InsertHoles (&e, *res, &holes);
     }
   /* delete holes if any left */
